example_test.go

```diff
@@ -117,6 +117,46 @@ func ExampleNew() {
 	// {"level":"info","msg":"This is an info log."}
 }
 
+func ExampleNew_textEncoder() {
+	// For more human-readable output in the console, use a TextEncoder.
+	textLogger := zap.New(zap.NewTextEncoder(
+		zap.TextNoTime(), // drop timestamps in tests.
+	))
+	textLogger.Info("This is a text log.")
+
+	// Output:
+	// [I] This is a text log.
+}
+
+func ExampleNew_textEncoderSingleCurlyBrace() {
+	textLogger := zap.New(zap.NewTextEncoder(
+		zap.TextNoTime(), // drop timestamps in tests.
+	))
+	textLogger.Info("Test 1.")
+	textLogger.Info("With fields", zap.String("f1", "v1"), zap.String("f2", "v2"))
+	textLogger.Info("With fields", zap.String("f1", "{"), zap.String("f2", "v2"))
+
+	// Output:
+	// [I] Test 1.
+	// [I] With fields f1=v1 f2=v2
+	// [I] With fields f1={ f2=v2
+}
+
+func ExampleNew_textEncoderNestedObject() {
+	m := func(kv zap.KeyValue) error {
+		kv.AddString("loggable", "yes")
+		kv.AddInt("number", 1)
+		return nil
+	}
+	textLogger := zap.New(zap.NewTextEncoder(
+		zap.TextNoTime(), // drop timestamps in tests.
+	))
+	textLogger.Info("With Marshaler", zap.Marshaler("m", zap.LogMarshalerFunc(m)))
+
+	// Output:
+	// [I] With Marshaler m={loggable=yes number=1}
```

Comment:
This sort of feels like it's less useful as an example but as a unit test. However, it doesn't look like the unit test framework is set up to do "end-to-end" (e.g. assert the actual output) and I wanted to show that I handles this case properly.

Is there a better place to put this?